v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -69,32 +73,52 @@ def from_ht(
             if dataset in ht.row:
                 # NB: handle an edge case (mito high constraint) where we annotate a bool from the reference dataset collection
                 selects[dataset] = (
-                    set(ht[dataset])
+                    {k: v.dtype for k, v in dict(ht[dataset]).items()}
                     if isinstance(ht[dataset], hl.StructExpression)
-                    else set()
+                    else {}
                 )
         return cls(paths, versions, enums, selects)
 
 
+def validate_selects_types(
```

Comment:
The idea here is to just ensure that any fields shared between the existing table and an "update" are identically typed.  

imo we're getting very close to needing a statically defined schema for these, but trying to squeeze that in was much messier.